nteract/hydrogen
# Convert to TypeScript

This converts the codebase to TypeScript

Requires:
https://github.com/nteract/kernelspecs/pull/35
https://github.com/nteract/jupyter-paths/pull/39

### The tests

The tests are all passing
https://github.com/aminya/hydrogen/actions/runs/733861352

### How to test the functionality
```
git clone https://github.com/aminya/hydrogen.git
cd hydrogen
git checkout typescript
apm install
npm run build
apm link .
atom .
```
Make sure to run `apm unlink .` after you're done testing, so you don't miss the official versions.


### How to review
You can go through the commits which include the mental logic behind the changes.

The first [commit conversion](https://github.com/nteract/hydrogen/pull/2063/commits/aed4482c9045aeec5de20abe2ce1ccaf6b31ab2a) is done automatically using [@khanacademy/flow-to-ts](https://github.com/Khan/flow-to-ts), and then the issues are fixed by hand. 

If you want to verify the changes make sure to disable the whitespace changes generated by the tool.
